include/pmacc/random/distributions/uniform/Uniform_double.hpp

```diff
@@ -0,0 +1,177 @@
+/* Copyright 2015-2018 Alexander Grund, Rene Widera
+ *
+ * This file is part of PMacc.
+ *
+ * PMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with PMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc/types.hpp"
+#include "pmacc/random/distributions/Uniform.hpp"
+#include "pmacc/random/distributions/uniform/Range.hpp"
+
+namespace pmacc
+{
+namespace random
+{
+namespace distributions
+{
+namespace detail
+{
+
+    /** Returns a random double value uniformly distributed in (0,1]
+     *
+     * The smallest created value is `2^-65` (~ `2.710505431213761*10^-20`)
```

Comment:
QUESTION:
is this distribution uniform in the possible numbers of float values or quasi-uniform the domain of real numbers?

Reply:
This creates random numbers without equal sized bins. I think this is what you mean with real numbers.

Reply:
Yes, the bins are not equal and thus their probability is not equal too - however both combined lead to a uniform distribution and that s fine - see comments below